llvm/lib/SandboxIR/Tracker.cpp

```diff
@@ -160,26 +160,28 @@ void CallBrInstSetIndirectDest::dump() const {
 }
 #endif
 
-SetVolatile::SetVolatile(Instruction *I, bool WasBool, Tracker &Tracker)
-    : IRChangeBase(Tracker), Inst(I) {
-  if (auto *Load = cast<llvm::LoadInst>(Inst)) {
+SetVolatile::SetVolatile(Instruction *I, Tracker &Tracker)
+    : IRChangeBase(Tracker) {
+  if (auto *Load = cast<llvm::LoadInst>(I)) {
     WasVolatile = Load->isVolatile();
     InstUnion = Load;
-  }
-  if (auto *Store = cast<llvm::StoreInst>(Inst)) {
+    assert(Load && "This isn't a LoadInst");
```

Comment:
This assertion is not needed, as if this is not a LoadInst we won't enter the `if`.

Reply:
remove the assert for both then?

Reply:
yes we don't need an assertion for either of them.

Reply:
I made sure to remove both of them.